pykickstart/commands/module.py

```diff
@@ -0,0 +1,115 @@
+from pykickstart.errors import KickstartParseError, formatErrorMsg
+from pykickstart.options import KSOptionParser
+from pykickstart.version import F28
+
+from pykickstart.i18n import _
+
+
+class F28_ModuleData(BaseData):
+    removedKeywords = BaseData.removedKeywords
+    removedAttrs = BaseData.removedAttrs
+
+    def __init__(self, *args, **kwargs):
+        BaseData.__init__(self, *args, **kwargs)
+        self.name = kwargs.get("name", "")
+        self.stream = kwargs.get("stream", "")
+
+    def __eq__(self, y):
+        if not y:
+            return False
+        return (self.name == y.name and self.stream == y.stream)
+
+    def __ne__(self, y):
+        return not self == y
+
+    def __str__(self):
+        retval = BaseData.__str__(self)
+        retval += "module --name=%s --stream=%s" % (self.name, self.stream)
+        return retval.strip() + "\n"
+
+
+class F28_Module(KickstartCommand):
+    """The module command makes it possible to manipulate modules.
+
+    (In this case we mean modules as introduced by the Fedora modularity initiative.)
+
+    A module is defined by a unique name and a stream id, where single module can
+    (and usually has) multiple available stream.
+
+    Streams will in most cases corresponds to stable releases of the given software
+    components (such as Node.js, Django, etc.) but there could be also other use cases,
+    such as a raw master stream or streams corresponding to an upcoming stable stream.
+
+    For more information see the Fedora modularity initiative documentation:
+    https://docs.pagure.org/modularity/
+    """
+
+    removedKeywords = KickstartCommand.removedKeywords
+    removedAttrs = KickstartCommand.removedAttrs
+
+    def __init__(self, writePriority=0, *args, **kwargs):
+        KickstartCommand.__init__(self, writePriority, *args, **kwargs)
+
+        self.moduleList = kwargs.get("moduleList", [])
+        self.op = self._getParser()
+
+    def __str__(self):
+        retval = ""
+
+        for module in self.moduleList:
+            retval += module.__str__()
+
+        return retval
+
+    def _getParser(self):
+        op = KSOptionParser(prog="module", version=F28, description="""
+                            Enable a module.""")
+        op.add_argument("--name", metavar="<module_name>", version=F28, required=True,
+                        help="""
+                        Name of the module to enable.""")
+        op.add_argument("--stream", metavar="<module_stream_name>", version=F28, required=True,
+                        help="""
+                        Name of the module stream to enable.""")
+
+        return op
+
+    def parse(self, args):
+        (ns, extra) = self.op.parse_known_args(args=args, lineno=self.lineno)
+
+        if len(extra) > 0:
+            msg = _("The enable module command does not take position arguments!")
+            raise KickstartParseError(formatErrorMsg(self.lineno, msg=msg), lineno=self.lineno)
```

Comment:
Function `formatErrorMsg` is deprecated now, use `KickstartParseError(msg, lineno=self.lineno)`.

Reply:
Fixed as well.